cluster-autoscaler/Makefile

```diff
@@ -104,4 +106,19 @@ container-arch-%: build-in-docker-arch-% make-image-arch-%
 test-in-docker: clean docker-builder
 	docker run ${RM_FLAG} -v `pwd`:/cluster-autoscaler/:Z autoscaling-builder:latest bash -c 'cd /cluster-autoscaler && go test -race ./... ${TAGS_FLAG}'
 
+juju-cli-images: $(addprefix juju-cli-image-arch-,$(ALL_ARCH))
```

Comment:
this make target can be used to build local images containing the juju cli, which can then be used as a base image in subsequent builds of the autoscaler image:
```
make juju-cli-images
# Push CLI Release
make TAG=dev-CLI BUILD_TAGS=juju GOARCH=amd64 BASEIMAGE=juju_cli:2.9.22-amd64 dev-release
```
You can also push a nonCLI release only containing the autoscaler without the juju CLI:
```
# Push Non CLI Release
make TAG=dev BUILD_TAGS=juju GOARCH=amd64 dev-release
```